File: ulina_server/src/discord/commands/remove_nation.rs
use crate::database::db;
use crate::discord::commands::shared::{default_data, edit_action};
use crate::discord::commands::shared::{Category, CommandData, CreateCommand, Interaction};
use crate::discord::helper::Helper;
use crate::error::Error;
use serenity::client::Context;
use serenity::http::Http;
use sqlx::query;

pub const DATA: CommandData = CommandData {
    admin_only: true,
    name: "edit-remove-nation",
    category: Category::EditNation,
};

pub fn create(command: &mut CreateCommand) -> &mut CreateCommand {
    default_data(command, &DATA).description("admin only - remove a nation")
}

pub async fn remove_nation(ctx: &Context, interaction: &Interaction) -> Result<(), Error> {
    let nation = edit_action(interaction, &DATA).await?;

    query!(
        "UPDATE Nation SET removed = true WHERE nationId = ?",
        nation.nationId
    )
    .execute(db())
    .await?;

    interaction
        .message(&ctx.http, |message| {
            message.content(format!("Successfully removed {}", nation.name))
        })
        .await;

    Ok(())
}

File: bot/commands/remove-nation.ts
import {CommandInteraction} from "discord.js";
import {UlinaCommand} from "../bot";
import {NationAndSocials, updateNation} from "../../server/database";
import {editData, editExecute, EditType} from "../nation-edit";
import {OkVoid} from "../../shared/result";
import {SlashCommandBuilder} from "@discordjs/builders";

async function execute(interaction: CommandInteraction, nation: NationAndSocials) {
    await updateNation({nationId: nation.nationId, removed: true});
    interaction.reply(`successfully removed ${nation.name}`);
    return OkVoid();
}

const edit: EditType = {
    adminOnly: true,
    execute,
    name: "remove nation"
}

module.exports = {
    data: editData(edit, new SlashCommandBuilder())
        .setDescription('admin only - remove a nation'),
    execute: editExecute(edit),
    category: "Edit Nation"
} as UlinaCommand;